packages/kerosene/src/datetime/getCalendarWeeks.ts

```diff
@@ -0,0 +1,57 @@
+import * as dateFns from "date-fns";
+import { range } from "lodash";
+import { DAYS_PER_WEEK, DayOfWeek } from ".";
+
+/**
+ * 7-tuple containing moments for each day of the calendar week
+ */
+export type CalendarWeek = [Date, Date, Date, Date, Date, Date, Date];
+
+/**
+ * 4-, 5-, or 6-tuple of calendar weeks which containing all of the days in a calendar month, plus the overhang (if any)
+ * at either end of the month
+ */
+export type CalendarWeeks =
+  | [CalendarWeek, CalendarWeek, CalendarWeek, CalendarWeek]
```

Comment:
are you sure that there's no instance in any country, anywhere, that's ever had a 3-week month?

Reply:
February is the shortest month, containing 28 days. 28 days divides exactly into 4 weeks. Though possibly something like the [Samoan time zone change of 2011](https://en.wikipedia.org/wiki/Time_in_Samoa#2011_time_zone_change) would cause headaches.

Reply:
do we care about those cases? datetime is such a pain -_-

Reply:
I'm willing to ignore this unless it becomes an issue